ibm/resource_ibm_satellite_dns_register.go

```diff
@@ -0,0 +1,94 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/IBM-Cloud/container-services-go-sdk/kubernetesserviceapiv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceIbmSatelliteDnsRegister() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIbmSatelliteDnsRegisterCreate,
+		ReadContext:   resourceIbmSatelliteDnsRegisterRead,
+		DeleteContext: resourceIbmSatelliteDnsRegisterDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ips": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceIbmSatelliteDnsRegisterCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	satClient, err := meta.(ClientSession).SatelliteClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	bmxSess, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	registerMultishiftClusterOptions := &kubernetesserviceapiv1.RegisterMultishiftClusterOptions{}
+
+	registerMultishiftClusterOptions.SetXAuthRefreshToken("Bearer " + bmxSess.Config.IAMAccessToken)
```

Comment:
Set the value to bmxSess.Config.IAMRefreshToken

Reply:
changed to bmxSess.Config.IAMRefreshToken